flow/cloud/cloudfoundry/cloudfoundry.py

```diff
@@ -825,8 +829,14 @@ def deploy(self, force_deploy=False, manifest=None):
             # for backup and force_deploy is used when you need to redeploy/replace an instance
             # that is currently running
             previous_versions = []
+            new_app_name = "{project_name}-{version}".format(project_name=self.config.project_name,
+                                                             version=self.config.version_number)
             for line in CloudFoundry.stopped_apps.splitlines():
-                previous_versions.append(line.decode("utf-8"))
+                if line.decode("utf-8").lower() != new_app_name.lower():
```

Comment:
I wonder if this would be better to save the splitlines() output to an array and then do a reduce to produce a new array, eliminate the print_msg in the else (unless we think it matters for the output, but I'm not sure it's useful.)